### PR TITLE
Nuke Docs via GithubActions + add compat for CairoMakie v0.11, v0.12

### LINKED ISSUE #157: Examples do not show in the docs
This is what I see:

image

@simone-silvestri

### LINKED ISSUE #161: Docs built
At the moment the docs are build with GitHub actions and also via buildkite-ci for examples.

But we can't have two ways of building the docs. When GitHub actions build docs without the examples then it pushes it and it overwrites the other docs. However the buildkite-CI only runs when the PR has a particular label... 

The workflow is not robust and results in situations like the one described in #157...

### PR DESCRIPTION
**Pull request by navidcy**
This PR closes #157. Also ~drops the previews and~ adds some more compat entries for CairoMakie. Closes #161.

### CODE REVIEW

**Review by simone-silvestri — APPROVED** [review]

**Comment by simone-silvestri** [comment]
Is there a way we can check the results without pushing the preview?

**Review by glwagner — APPROVED** [review]

**Comment by navidcy** [comment]
> Is there a way we can check the results without pushing the preview?

No.

We should enable the preview when we want for particular PRs... We don't want previews for every random PR because things will get out of hand.

**Comment by glwagner** [comment]
> Is there a way we can check the results without pushing the preview?

Also I don't think we should use the example to assess PRs, they take way too long to run. We have a lot of development to do so we need a way to assess PRs more quickly...

We can put problems that are similar to the examples in the docs, and create plots from them if we want (ClimaAtmos does something similar).

**Comment by glwagner** [comment]
> > Is there a way we can check the results without pushing the preview?
> 
> No.
> 
> We should enable the preview when we want for particular PRs... We don't want previews for every random PR because things will get out of hand.

An alternative though is if @simone-silvestri wants to take responsibility for deleting the old push previews right?

**Comment by simone-silvestri** [comment]
Do you mean deleting the previews after the PR is merged? That could be a nice idea. Maybe we can merge this PR and I can investigate if it is possible to do this in an automatic manner

**Comment by glwagner** [comment]
> Do you mean deleting the previews after the PR is merged? That could be a nice idea. Maybe we can merge this PR and I can investigate if it is possible to do this in an automatic manner

We already attempt to do it automatically, but it doesn't always work. So previously, @navidcy took responsibility for monitoring the repo and going in and manually deleting things when it filled up (similar to what I do with the buildkite builds). But @navidcy eventually decided it was too much work. However you can decide to take this responsibility if you want to keep the previews. I don't think you'll be able to completely automate the process.

**Comment by simone-silvestri** [comment]
Ok, I can take responsibility for that and see how it goes.

**Comment by navidcy** [comment]
There is a way to do it automatically if the `gh-pages` are in the same repo. It's documented in the documentation of Documenter.jl and actually the example there is from a CliMA repo!! However, it doesn't work for when the Docs live in a different repo. I tried a lot and also asked CliMA to help me to make it work for when the docs are deployed in a different repo but I never managed to do it. If you figure how to do that it'd be great!!

**Comment by navidcy** [comment]
Btw, why are the examples still haven't run?

**Comment by navidcy** [comment]
https://buildkite.com/clima/climaocean-examples/builds/355

??

**Comment by simone-silvestri** [comment]
The `Run examples` label triggers the build, without that the examples are not run. I have added the label but you have to commit again to trigger the pipeline

**Comment by navidcy** [comment]
https://buildkite.com/clima/climaocean-examples/builds/356#019176f1-b195-44bb-8114-d68491646a86/207-1012

Are the files id Dropbox still there?

**Comment by simone-silvestri** [comment]
I don't think they were deleted. maybe restarting the build will download them correctly?

**Comment by navidcy** [comment]
Any idea what's happening here?
https://buildkite.com/clima/climaocean-examples/builds/364#01917bd9-f3d5-4550-8c97-3f6d79db3fbf/207-1271

**Comment by simone-silvestri** [comment]
out of bounds error since `v` has an extra point in y, should be fixed by doing
```julia
T1 = interior(T[1], :, :, 1)
v1 = interior(v[1], :, :, 1)
landᵃᶜᵃ = T1 .== 0
landᵃᶠᵃ = v1 .== 0
for n = 1:Nt
      ui = interior(u[n], :, :, 1)
      vi = interior(v[n], :, :, 1)
      Ti = interior(T[n], :, :, 1)
      ei = interior(e[n], :, :, 1)
      ui[landᵃᶜᵃ] .= NaN
      vi[landᵃᶠᵃ] .= NaN
      Ti[landᵃᶜᵃ] .= NaN
      ei[landᵃᶜᵃ] .= NaN
end
```

**Comment by glwagner** [comment]
> out of bounds error since `v` has an extra point in y, should be fixed by doing
> 
> ```julia
> T1 = interior(T[1], :, :, 1)
> v1 = interior(v[1], :, :, 1)
> landᵃᶜᵃ = T1 .== 0
> landᵃᶠᵃ = v1 .== 0
> for n = 1:Nt
>       ui = interior(u[n], :, :, 1)
>       vi = interior(v[n], :, :, 1)
>       Ti = interior(T[n], :, :, 1)
>       ei = interior(e[n], :, :, 1)
>       ui[landᵃᶜᵃ] .= NaN
>       vi[landᵃᶠᵃ] .= NaN
>       Ti[landᵃᶜᵃ] .= NaN
>       ei[landᵃᶜᵃ] .= NaN
> end
> ```
> 
> another option is
> 
> ```julia
> T1 = interior(T[1], :, :, 1)
> land = T1 .== 0
> for n = 1:Nt
>       ui = interior(u[n], :, :, 1)
>       vi = interior(v[n], :, 1:Ny, 1)
>       Ti = interior(T[n], :, :, 1)
>       ei = interior(e[n], :, :, 1)
>       ui[land] .= NaN
>       vi[land] .= NaN
>       Ti[land] .= NaN
>       ei[land] .= NaN
> end
> ```

Messy for an example...

**Comment by navidcy** [comment]
I think the example should compute the speed on Center and then apply the land mask on the speed.

**Comment by glwagner** [comment]
> I think the example should compute the speed on Center and then apply the land mask on the speed, not on u and v. That would be simpler at this stage, right?

Yeah good idea

**Comment by navidcy** [comment]
Otherwise write a short method that applies the land mask to every FieldTimeSeries?

E.g.,

```Julia
function mask_land!(f)
    times = f.times
    Nt = length(f)

    f1 = interior(f[1], :, :, 1)
    land = f1 .== 0

    for n = 1:Nt
          fi = interior(f[n], :, :, 1)
          fi[land] .= NaN
    end

    return nothing
end

for f in (u, v, T, e):
    mask_land!(f)
end
```

**Comment by navidcy** [comment]
Actually the method I suggested could be part of the source code?

**Comment by simone-silvestri** [comment]
Another option is to use `mask_immersed_field!`

```julia
using Oceananigans.ImmersedBoundaryConditions: mask_immersed_field!
mask_immersed_field!(field, NaN)
```

**Comment by glwagner** [comment]
> Another option is to use `mask_immersed_field!`
> 
> ```julia
> using Oceananigans.ImmersedBoundaryConditions: mask_immersed_field!
> mask_immersed_field!(field, NaN)
> ```

Hmm right that is probably the right solution. We should also extend that to `FieldTimeSeries`. And maybe change the name because its not just for fields: `mask_immersed_cells!` or just `mask_immersed!` or something.

What I actually meant was to avoid the need to mask immersed regions altogether. It's just a hassle for visualization, somehow it should happen automatically.

A few thoughts:

1. We still need the ability to apply a mask before output.
2. We could extend the visualization capabilities to automatically turn immersed cells to NaN for visualization?
3. Let's not conflate immersed regions with "land". Its tempting but we want to retain generality.

**Comment by navidcy** [comment]
The Makie extension in Oceananigans now masks automatically. Thanks @glwagner ;)
We can bump a patch release in Oceananigans and use that version here.

**Comment by navidcy** [comment]
@simone-silvestri any idea what this error means?

https://buildkite.com/clima/climaocean-examples/builds/372#01918cbe-13f1-4d36-8fc9-0eb62a00b8ca/207-315

I can't seem to find any `target_times` anywhere in the repo so I'm confused.

**Comment by navidcy** [comment]
@simone-silvestri do you have any insight why the examples didn't run the last time? I don't see any error... I see a segfault... :(

**Comment by glwagner** [comment]
I think it segfaulted while making the movie right?

**Comment by glwagner** [comment]
Why don't you just shorten the simulation a lot so it runs faster

**Comment by navidcy** [comment]
I just didn't wanna modify the example from what @simone-silvestri had. Happy to do so tho

**Comment by glwagner** [comment]
Is the goal to inspect the movie to determine whether the simulation is ok, or simply to ensure the example runs? I'd say the second, we shouldn't use the examples as validation. They are a really inefficient way of doing validation I think and fail for all sorts of reasons (not because the physics is bad or something).

**Comment by navidcy** [comment]
Another segfault...
How do we proceed? :(

**Comment by navidcy** [comment]
> Is the goal to inspect the movie to determine whether the simulation is ok, or simply to ensure the example runs? I'd say the second, we shouldn't use the examples as validation. They are a really inefficient way of doing validation I think and fail for all sorts of reasons (not because the physics is bad or something).

The second. To ensure that the movie was actually produced mainly..

**Comment by simone-silvestri** [comment]
The segfault happens within these lines
```julia
  | │ u = FieldTimeSeries("surface.jld2", "u"; backend = OnDisk())
  | │ v = FieldTimeSeries("surface.jld2", "v"; backend = OnDisk())
  | │ T = FieldTimeSeries("surface.jld2", "T"; backend = OnDisk())
  | │ e = FieldTimeSeries("surface.jld2", "e"; backend = OnDisk())
  | │
  | │ times = u.times
  | │ Nt = length(times)
  | │
  | │ iter = Observable(Nt)
  | │
  | │ Ti = @lift T[$iter]
  | │ ei = @lift e[$iter]
  | │
  | │ si = @lift begin
  | │      s = Field(sqrt(u[$iter]^2 + v[$iter]^2))
  | │      compute!(s)
  | │ end
  | │
  | │ fig = Figure(size = (800, 400))
  | │ ax = Axis(fig[1, 1])
  | │ hm = heatmap!(ax, si, colorrange = (0, 0.5), colormap = :deep)
  | │ cb = Colorbar(fig[0, 1], hm, vertical = false, label = "Surface speed (ms⁻¹)")
  | │ hidedecorations!(ax)
```
you can try running the example locally using `julia --check-bounds=yes` to debug it. There is probably no need to even run the simulation, just having it save the initial condition down to test this code block.

**Comment by glwagner** [comment]
Next step is to reproduce the error outside of CI

**Comment by navidcy** [comment]
I tried to run on tartarus and bumped onto this error:

```Julia
julia> set!(model, T=ECCOMetadata(:temperature; date), S=ECCOMetadata(:salinity; date))
ERROR: ArgumentError: Could not find the username for https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/daily/. Please provide a username in the ECCO_USERNAME environment variable.
Stacktrace:
  [1] download_dataset!(metadata::ECCOMetadata{DateTimeProlepticGregorian, ClimaOcean.DataWrangling.ECCO.ECCO2Daily}; url::String)
    @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ecco_metadata.jl:156
  [2] download_dataset!
    @ ~/ClimaOcean.jl/src/DataWrangling/ecco_metadata.jl:144 [inlined]
  [3] ecco_field(metadata::ECCOMetadata{…}; architecture::GPU, horizontal_halo::Tuple{…}, filename::String)
    @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:140
  [4] ecco_field
    @ ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:133 [inlined]
  [5] ecco_mask(metadata::ECCOMetadata{…}, architecture::GPU; minimum_value::Float32, maximum_value::Float32, filename::String)
    @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:204
  [6] ecco_mask(metadata::ECCOMetadata{DateTimeProlepticGregorian, ClimaOcean.DataWrangling.ECCO.ECCO2Daily}, architecture::GPU)
    @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:199
  [7] set!(field::Field{…}, ecco_metadata::ECCOMetadata{…}; kw::@Kwargs{})
    @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:299
  [8] set!(field::Field{…}, ecco_metadata::ECCOMetadata{…})
    @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:295
  [9] apply_regionally!(::Function, ::Field{…}, ::Vararg{…}; kwargs::@Kwargs{})
    @ Oceananigans.Utils ~/.julia/packages/Oceananigans/Hkk5J/src/Utils/multi_region_transformation.jl:121
 [10] apply_regionally!
    @ ~/.julia/packages/Oceananigans/Hkk5J/src/Utils/multi_region_transformation.jl:118 [inlined]
 [11] macro expansion
    @ ~/.julia/packages/Oceananigans/Hkk5J/src/Utils/multi_region_transformation.jl:206 [inlined]
 [12] #set!#40
    @ ~/.julia/packages/Oceananigans/Hkk5J/src/Models/HydrostaticFreeSurfaceModels/set_hydrostatic_free_surface_model.jl:60 [inlined]
 [13] top-level scope
    @ REPL[29]:1
Some type information was truncated. Use `show(err)` to see complete types.
```

It must be something with the credentials that the buildkite env is aware of? I couldn't figure out where these are... Are they part of a repo's secret (which I can't see)?

**Comment by glwagner** [comment]
> I tried to run on tartarus and bumped onto this error:
> 
> ```julia
> julia> set!(model, T=ECCOMetadata(:temperature; date), S=ECCOMetadata(:salinity; date))
> ERROR: ArgumentError: Could not find the username for https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/daily/. Please provide a username in the ECCO_USERNAME environment variable.
> Stacktrace:
>   [1] download_dataset!(metadata::ECCOMetadata{DateTimeProlepticGregorian, ClimaOcean.DataWrangling.ECCO.ECCO2Daily}; url::String)
>     @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ecco_metadata.jl:156
>   [2] download_dataset!
>     @ ~/ClimaOcean.jl/src/DataWrangling/ecco_metadata.jl:144 [inlined]
>   [3] ecco_field(metadata::ECCOMetadata{…}; architecture::GPU, horizontal_halo::Tuple{…}, filename::String)
>     @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:140
>   [4] ecco_field
>     @ ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:133 [inlined]
>   [5] ecco_mask(metadata::ECCOMetadata{…}, architecture::GPU; minimum_value::Float32, maximum_value::Float32, filename::String)
>     @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:204
>   [6] ecco_mask(metadata::ECCOMetadata{DateTimeProlepticGregorian, ClimaOcean.DataWrangling.ECCO.ECCO2Daily}, architecture::GPU)
>     @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:199
>   [7] set!(field::Field{…}, ecco_metadata::ECCOMetadata{…}; kw::@Kwargs{})
>     @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:299
>   [8] set!(field::Field{…}, ecco_metadata::ECCOMetadata{…})
>     @ ClimaOcean.DataWrangling.ECCO ~/ClimaOcean.jl/src/DataWrangling/ECCO.jl:295
>   [9] apply_regionally!(::Function, ::Field{…}, ::Vararg{…}; kwargs::@Kwargs{})
>     @ Oceananigans.Utils ~/.julia/packages/Oceananigans/Hkk5J/src/Utils/multi_region_transformation.jl:121
>  [10] apply_regionally!
>     @ ~/.julia/packages/Oceananigans/Hkk5J/src/Utils/multi_region_transformation.jl:118 [inlined]
>  [11] macro expansion
>     @ ~/.julia/packages/Oceananigans/Hkk5J/src/Utils/multi_region_transformation.jl:206 [inlined]
>  [12] #set!#40
>     @ ~/.julia/packages/Oceananigans/Hkk5J/src/Models/HydrostaticFreeSurfaceModels/set_hydrostatic_free_surface_model.jl:60 [inlined]
>  [13] top-level scope
>     @ REPL[29]:1
> Some type information was truncated. Use `show(err)` to see complete types.
> ```
> 
> It must be something with the credentials that the buildkite env is aware of? I couldn't figure out where these are... Are they part of a repo's secret (which I can't see)?

Yes, part of the secret I think. To reproduce the example you have to follow the instructions in the error and obtain credentials to download the ECCO data. I don't think there's a way around this. We can't make the credentials public.

**Comment by glwagner** [comment]
There's more instructions here:

https://github.com/CliMA/ClimaOcean.jl/blob/ca9385ee298791541da004c2e23bef5cbf1fb742/src/DataWrangling/ECCO/README.md

which is part of https://github.com/CliMA/ClimaOcean.jl/pull/169

**Comment by glwagner** [comment]
@navidcy I will take a look at this and get it working.

@simone-silvestri some feedback on your example --- I think there is too much text. We want to write the code as much as possible to be self-documenting and also let the `show` methods provide as much info as possible. I'm going to make it a little more concise.

**Comment by glwagner** [comment]
Here are some more questions --- in Oceananigans we always use `n` to refer to the time index. But the near global example uses `iter`. The reason I avoided `iter` is because it is easy to confuse the time index with the iteration of the simulation, when they are different (unless we save every iteration). Do we want to keep `iter`, or are we ok with the Oceananigans convention of `n`? I'm happy to rethink the convention if that seems right. Also much of the notation uses "i" intsead of "n", like `Ti` for example. @simone-silvestri do you have any comments about that?

**Comment by simone-silvestri** [comment]
Right, `iter` might be confused with iteration. `n` sounds good as an index.

**Comment by simone-silvestri** [comment]
> The time step wizard is not going to work out of the box for ClimaOcean stuff --- we have to adapt it for simulations of `OceanSeaIceModel`

Let's just remove it for now until we have a working RK3 for the hydrostatic model

**Comment by glwagner** [comment]
> > The time step wizard is not going to work out of the box for ClimaOcean stuff --- we have to adapt it for simulations of `OceanSeaIceModel`
> 
> Let's just remove it for now until we have a working RK3 for the hydrostatic model

boom

**Comment by glwagner** [comment]
@simone-silvestri I can't find "Payne (1982)". I searched "payne 1982 albedo" and found nothing:

https://scholar.google.com/scholar?hl=en&as_sdt=0%2C13&q=payne+1982+albedo&btnG=

I think it could be maybe not a good idea to document defaults in the examples. They will go stale easily and we might not catch them...

**Comment by navidcy** [comment]
weird CUDA error -- first time I've seen this
https://buildkite.com/clima/climaocean-ci/builds/1319#0191c6d2-dad6-42bb-8e55-016f31e5cab0/206-472 ?

File: docs/Project.toml
Comment (glwagner) [inline docs/Project.toml:4]: we're supposed to dev ClimaOcean right?

**Comment by glwagner** [comment]
> weird CUDA error -- first time I've seen this https://buildkite.com/clima/climaocean-ci/builds/1319#0191c6d2-dad6-42bb-8e55-016f31e5cab0/206-472 ?

It's a machine / setup issue with whatever node its running on

File: docs/Project.toml
Comment (navidcy) [reply docs/Project.toml:4]: yes, that shouldn't be there... if I committed it, it was by accident

**Comment by navidcy** [comment]
> > weird CUDA error -- first time I've seen this https://buildkite.com/clima/climaocean-ci/builds/1319#0191c6d2-dad6-42bb-8e55-016f31e5cab0/206-472 ?
> 
> It's a machine / setup issue with whatever node its running on

I found it strange that it appeared out of the blue as we didn't change anything on the build kite setup. But perhaps it had nothing to do with the PR....

**Comment by glwagner** [comment]
> > > weird CUDA error -- first time I've seen this https://buildkite.com/clima/climaocean-ci/builds/1319#0191c6d2-dad6-42bb-8e55-016f31e5cab0/206-472 ?
> > 
> > 
> > It's a machine / setup issue with whatever node its running on
> 
> I found it strange that it appeared out of the blue as we didn't change anything on the build kite setup. But perhaps it had nothing to do with the PR....

Yes these crop up intermittently it seems

**Comment by navidcy** [comment]
I can't reproduce the errors because I don't have credentials to download the bathymetry -- I don't know how build kite does it. Probably it has a secret saved somewhere secretly?

**Comment by glwagner** [comment]
@navidcy see this markdown for instructions on obtaining ECCO credentials. They are available to anyone. This is part of #169 

This sort of thing is going to be common with ClimaOcean --- every dataset we add will probably require obtaining external credentials and storing the secret in an environment variable. So if the process isn't clear, let's make the docs and errors better.

**Comment by navidcy** [comment]
> @navidcy see this markdown for instructions on obtaining ECCO credentials: https://github.com/CliMA/ClimaOcean.jl/tree/glw/clean-up-ECCO/src/DataWrangling/ECCO
> 
> The credentials are available to anyone. The markdown doc is part of #169
> 
> This sort of thing is going to be common with ClimaOcean --- every dataset we add will probably require obtaining external credentials and storing the secret in an environment variable. So if the process isn't clear, let's make the docs and errors better.
> 
> And yes, we have stored the credentials as a secret (I believe this is on the github repo itself, which I think is the best practice for this kind of thing). But recommend another workflow if you know of a better one.

Storing as a GitHub secret is great. I was just wondering because I didn't know so I couldn't figure out how to reproduce this locally. Thanks for the info above!

**Comment by glwagner** [comment]
> > @navidcy see this markdown for instructions on obtaining ECCO credentials: https://github.com/CliMA/ClimaOcean.jl/tree/glw/clean-up-ECCO/src/DataWrangling/ECCO
> > The credentials are available to anyone. The markdown doc is part of #169
> > This sort of thing is going to be common with ClimaOcean --- every dataset we add will probably require obtaining external credentials and storing the secret in an environment variable. So if the process isn't clear, let's make the docs and errors better.
> > And yes, we have stored the credentials as a secret (I believe this is on the github repo itself, which I think is the best practice for this kind of thing). But recommend another workflow if you know of a better one.
> 
> Storing as a GitHub secret is great. I was just wondering because I didn't know so I couldn't figure out how to reproduce this locally. Thanks for the info above!

Possibly we should include links within the example itself to help people if they run into issues? Or maybe the errors will be enough?

**Comment by navidcy** [comment]
> Possibly we should include links within the example itself to help people if they run into issues? Or maybe the errors will be enough?

Definitely!

**Comment by navidcy** [comment]
ERROR: LoadError: LoadError: Number of threads per block exceeds kernel limit (1024 > 896).
in expression starting at /central/scratch/esm/slurm-buildkite/climaocean-examples/729/climaocean-examples/docs/src/literated/near_global_ocean_simulation.md:2
when executing the following code block from inputfile `/central/scratch/esm/slurm-buildkite/climaocean-examples/729/climaocean-examples/examples/near_global_ocean_simulation.jl`

```julia
date = DateTimeProlepticGregorian(1993, 1, 1)
set!(ocean.model, T=ECCOMetadata(:temperature; dates=date), S=ECCOMetadata(:salinity; dates=date))
```


https://buildkite.com/clima/climaocean-examples/builds/729#0192e4d0-1ff4-4c6d-9570-c163ce343a40/189-1417


Any idea what that is and how to avoid it? On tartarus that line went through fine!

**Comment by navidcy** [comment]
I have some progress!

```Julia
julia> Tn
Observable(1440×600×1 Field{Center, Center, Center} on ImmersedBoundaryGrid on CPU
├── grid: 1440×600×40 ImmersedBoundaryGrid{Float64, Periodic, Bounded, Bounded} on CPU with 7×7×7 halo
├── boundary conditions: FieldBoundaryConditions
│   └── west: Periodic, east: Periodic, south: ZeroFlux, north: ZeroFlux, bottom: Nothing, top: Nothing, immersed: ZeroFlux
├── indices: (:, :, 40:40)
└── data: 1454×614×1 OffsetArray(::Array{Float32, 3}, -6:1447, -6:607, 40:40) with eltype Float32 with indices -6:1447×-6:607×40:40
    └── max=31.2508, min=-1.96045, mean=0.476186)


julia> heatmap(Tn)

[1626384] signal (11.128): Segmentation fault
in expression starting at REPL[56]:1
Segmentation fault (core dumped)
```

**Comment by glwagner** [comment]
> ERROR: LoadError: LoadError: Number of threads per block exceeds kernel limit (1024 > 896). in expression starting at /central/scratch/esm/slurm-buildkite/climaocean-examples/729/climaocean-examples/docs/src/literated/near_global_ocean_simulation.md:2 when executing the following code block from inputfile `/central/scratch/esm/slurm-buildkite/climaocean-examples/729/climaocean-examples/examples/near_global_ocean_simulation.jl`
> 
> ```julia
> date = DateTimeProlepticGregorian(1993, 1, 1)
> set!(ocean.model, T=ECCOMetadata(:temperature; dates=date), S=ECCOMetadata(:salinity; dates=date))
> ```
> 
> https://buildkite.com/clima/climaocean-examples/builds/729#0192e4d0-1ff4-4c6d-9570-c163ce343a40/189-1417
> 
> Any idea what that is and how to avoid it? On tartarus that line went through fine!

I think you need to consider a smaller problem / grid

**Pull request merged**
